src/convex/ConvexAvatarMultiToken.sol

```diff
@@ -579,6 +596,38 @@ contract ConvexAvatarMultiToken is BaseAvatar, ConvexAvatarUtils, PausableUpgrad
         }
     }
 
+    /// @notice Withdraws curve lp token into underlyings proportionally
+    /// @param _lpToken Curve lp address
+    /// @param _lpAmount Amount of bpt to be withdrawn
+    function _withdrawLpToUnderlyings(address _lpToken, uint256 _lpAmount) internal {
+        ICurvePool pool = ICurvePool(META_REGISTRY.get_pool_from_lp_token(_lpToken));
```

Comment:
is the meta pool all inclusive? should we for loop/fuzz test all the different types of curve lp tokens?

Reply:
added the suggestion to catch zero addr in 83948f4d3493de3402ed788319c66815580a033c, will add also fuzzing test to see for many pids within the convex booster how reliable the metaregistry can be

Reply:
added fuzz test in ed21c3513b9e2394fb7c3845dbe85f899616634b, run it locally with the following cmd:

`forge test --match-path test/convex/ConvexAvatarMultiTokenTest.t.sol --match-test "testFuzz*"`

i left out explicitly the fuzzing test in the *forge.yml* with `forge test --no-match-test "testFuzz*"`, since it can really take a while for the pipeline to complete. Also, set runs to 50, since the pid is bounded from 0 to the `poolLength` of the convex booster and by default runs are set at 256. I think 50/148 covers quite a bit the checks in the meta-registry, but feel free to set to default in your local run if you wish to be more exhaustive.

one realisation during the fuzzing is that currently the avatar does not accept eth, so if one of the underlying coins is `0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE` from the curve pool, will raise a revert, since we cannot leverage in every pool a direct transfer of the tokens into the owner. was thinking if we should add a fallback to allow the contract receiving eth so it can send it to within `_withdrawLpToUnderlyings`, but it will need quite a bit of change cause on the way that `__BaseAvatar_init`& `OwnableUpgradeable` is set. A priori do not see any concern in adding the fallback, but would like first to discuss.

e.g:

```
Logs:
  Bound Result 128
  PID run for 128
  LP address at 0xf43211935C781D5ca1a41d2041F397B8A7366C7A
  Pool address found by metaregistry at 0xa1F8A6807c402E4A15ef4EBa36528A3FED24E577
```

Another realisation is that some pools in *coins* are expecting a [uint256](https://etherscan.io/address/0xa1F8A6807c402E4A15ef4EBa36528A3FED24E577#code#L79) while other [int256](https://etherscan.io/address/0xA2B47E3D5c44877cca798226B7B8118F9BFb7A56#code#L54), the abi structure of pools gets bit messy to capture a smooth logic on-chain. In the fuzz test i used a try/catch approach, but not sure if it is the most reliable way to handle it. 

Other finding during the fuzz was those pools with 4 tokens, simply for more robust method for withdrawals i added support for 2, 3 and 4 token pools, so we avoid that corner case revert of not pointing to the right signature.

Regarding the meta-registry failing with a zero_address so far in my local fuzz runs was only pid 48, where the ["lp"](https://etherscan.io/address/0xB15fFb543211b558D40160811e5DcBcd7d5aaac9) somehow does not look like a curve lp at all. That case will revert now in L606 with `PoolNotFoundInMetaregistry`.

Lastly, there is many cases whee the underlying like TrueUSD or alike has some whitelisting logic in the internal _transfer method, which makes tricker to do the fuzzing test smooth so far without some unexpected revert, quite interesting how many peculiarities there is in the curve pools.

Overall, while to some extent i though that the fuzzing may be overkilling it raises quite interesting findings during the exercise.

thoughts?

Reply:
will holding an eth pool make an emergency call like `withdrawAll(true)` revert? if so, i def think support for withdrawing to native eth should be added.

if not, at least we should document the fact that it is not supported and what that means for the emergency withdrawal call.

Reply:
yes, it will revert in that case

Reply:
First of all, I do not think it is an overkill, better to lever the tool and test even more unforeseen scenarios now; good call and good job on the fuzzing!

About accepting ETH, I know that for the time being there's no plans of farming ETH based LPs but I can see it being the case for when eBTC launches. Of course, that's too far into the future and uncertain. At the same time, we could always develop a variation to the emergency withdraw script for these kinds of LPs that performs a normal withdraw and removes liquidity atomically via the multisig. This would have to be executed from the Treasury Vault which defeats the purpose of agility. Given what it will mean in terms of additional dev work and security reviews I say I am with Gos, let's document this and develop the required scripts for proper handling as needed.

As per the signature incompatibilities, I think that as long as we document which pools we can't handle, it should be enough. I don't think that the work that would be required to adjust to all different scenarios is justified for the time being.

Reply:
just as a note: if we decide to go ahead with this func but choose not to support eth pools, i think we should make sure it doesnt revert even if there was an eth pool present.